docs/GPU-Targets.rst

```diff
@@ -4,39 +4,42 @@ GPU SDK Correspondence and Device Targeting Table
+In general, OpenCL CPU backends are quite slow, and should be used for testing and debugging only.
 
 You can find below a table of correspondence:
 
 +---------------------------+-----------------+-----------------+-----------------+--------------+
 | SDK                       | CPU Intel/AMD   | GPU Intel       | GPU AMD         | GPU NVIDIA   |
 +===========================+=================+=================+=================+==============+
-| `Intel SDK for OpenCL`_   | Supported       | Supported \*    | Supported       | Untested     |
+| `Intel SDK for OpenCL`_   | Supported       | Supported       | Not Supported   | Not Supported|
 +---------------------------+-----------------+-----------------+-----------------+--------------+
-| AMD APP SDK \*\*\*        | Supported       | Untested \*     | Supported       | Fails        |
+| AMD APP SDK \*            | Supported       | Not Supported   | Supported       | Not Supported|
 +---------------------------+-----------------+-----------------+-----------------+--------------+
-| `NVIDIA CUDA Toolkit`_    | Fails    \*\*   | Fails    \*\*   | Fails    \*\*   | Supported    |
+| `NVIDIA CUDA Toolkit`_    | Not Supported   | Not Supported   | Not Supported   | Supported    |
 +---------------------------+-----------------+-----------------+-----------------+--------------+
 
 Legend:
 
--  \* Not usable directly.
--  \*\* Reported as unsupported in public forums.
-- \*\*\* AMD has decided to drop the support for APP SDK and deleted all links to installation packages. You can download the installation package for Linux from `our GitHub repo`_.
-
-AMD GPUs using Intel SDK for OpenCL is not a typo, nor AMD APP SDK compatibility with CPUs.
+- \* AMD APP SDK is deprecated. On Windows, OpenCL is included in AMD graphics driver. On Linux, newer generation AMD cards are supported by the `ROCm`_ driver.
 
 --------------
 
-Targeting Table
-===============
+Query OpenCL Devices in Your System
+===================================
+
+Your system might have multiple GPUs from different vendors ("platforms") installed. Setting up LightGBM GPU device requries two parameters: OpenCL Platform ID (``gpu_platform_id``) and OpenCL Device ID (``gpu_device_id``). Generally speaking, each vendor provides a OpenCL Platform, and devices from the same vendor have different device IDs under that platform. For example, if your system has a Intel Integrated GPU and two discrete GPUs from AMD, you will have two OpenCL platforms (with ``gpu_platform_id=0`` and ``gpu_platform_id=1``). If the platform 0 is Intel, it has one device (``gpu_device_id=0``) representing the Intel GPU; If the platform 1 is AMD, it has two devices (``gpu_device_id=0``, ``gpu_device_id=1``) representing the two AMD GPUs. If you have a discrete GPU by AMD/NVIDIA and an integrated GPU by Intel, make sure to select the correct ``gpu_platform_id`` to use the discrete GPU as it usually provides better performance.
+
```

Comment:
I think it's good idea to make some cross links for parameters list.
In `GPU-Targets.rst`:
```
... requries two parameters: `OpenCL Platform ID <./Parameters.rst#gpu_platform_id>`__ (``gpu_platform_id``) and `OpenCL Device ID <./Parameters.rst#gpu_device_id>`__ (``gpu_device_id``).
```
and in [`config.h`](https://github.com/Microsoft/LightGBM/blob/master/include/LightGBM/config.h) (after modifying config.h please run `python ../../helper/parameter_generator.py`):
```
// desc = OpenCL platform ID. Usually each GPU vendor exposes one OpenCL platform
// desc = **Note**: refer to `GPU Targets <./GPU-Targets.rst#query-opencl-devices-in-your-system>`__ for more details

...

// desc = OpenCL device ID in the specified platform. Each GPU in the selected platform has a unique device ID
// desc = **Note**: refer to `GPU Targets <./GPU-Targets.rst#query-opencl-devices-in-your-system>`__ for more details
```